Week3/homework/step2-3.js

```diff
@@ -2,41 +2,36 @@
 
 // Use a 'for' loop
 function repeatStringNumTimesWithFor(str, num) {
-  // eslint-disable-next-line prefer-const
   let result = '';
-
-  // Replace this comment and the next line with your code
-  console.log(str, num, result);
-
+  for (num; num > 0; num--) {
+    result += str;
+  }
   return result;
 }
-
 console.log('for', repeatStringNumTimesWithFor('abc', 3));
 
 // Use a 'while' loop
 function repeatStringNumTimesWithWhile(str, num) {
-  // eslint-disable-next-line prefer-const
   let result = '';
-
-  // Replace this comment and the next line with your code
-  console.log(str, num, result);
-
+  while (num > 0) {
+    result += str;
+    num--;
+  }
   return result;
 }
-
 console.log('while', repeatStringNumTimesWithWhile('abc', 3));
 
 // Use a 'do...while' loop
 function repeatStringNumTimesWithDoWhile(str, num) {
-  // eslint-disable-next-line prefer-const
   let result = '';
-
-  // Replace this comment and the next line with your code
-  console.log(str, num, result);
-
+  if (num > 0) {
+    do {
+      result += str;
+      num--;
+    } while (num > 0);
+  }
```

Comment:
Any reason why you added this if check here in the do-while loop and not for or while loop? 🙂

Reply:
Because , for and while loops check the conditions before executing the code. In do - while loop, it executes the condtions atleast once, therefore I wanted to add if condition for preventing do execute the code.